spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/util/SpanNameUtil.java

```diff
@@ -29,12 +29,22 @@ public static String toLowerHyphen(String name) {
 		StringBuilder result = new StringBuilder();
 		for (int i = 0; i < name.length(); i++) {
 			char c = name.charAt(i);
-			if (c >= 'A' && c <= 'Z') {
-				result.append('-').append((char) (c + 'a' - 'A'));
+			if (i > 0 && isUpercase(c)) {
```

Comment:
oh I think I see why. bc this assumed the input was in lower-camel, but it is now used for upper-camel, too.

another way could be to special case the first character outside the loop and initialize `int i = 1`, leaving the rest alone.